pkg/api/pod/warnings.go

```diff
@@ -172,6 +172,9 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		if v.PhotonPersistentDisk != nil {
 			warnings = append(warnings, fmt.Sprintf("%s: non-functional in v1.16+", fieldPath.Child("spec", "volumes").Index(i).Child("photonPersistentDisk")))
 		}
+		if v.ScaleIO != nil {
```

Comment:
please add a unit test exercising this warning

Reply:
Done